Content.Server/Wieldable/WieldableSystem.cs

```diff
@@ -0,0 +1,280 @@
+﻿using Content.Server.DoAfter;
+using Content.Server.Hands.Components;
+using Content.Server.Hands.Systems;
+using Content.Server.Items;
+using Content.Server.Weapon.Melee;
+using Content.Server.Wieldable.Components;
+using Content.Shared.Hands;
+using Content.Shared.Hands.Components;
+using Content.Shared.Interaction;
+using Content.Shared.Notification.Managers;
+using Content.Shared.Throwing;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+
+namespace Content.Server.Wieldable
+{
+    public class WieldableSystem : EntitySystem
+    {
+        [Dependency] private readonly DoAfterSystem _doAfter = default!;
+        [Dependency] private readonly HandVirtualItemSystem _virtualItemSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<WieldableComponent, UseInHandEvent>(OnUseInHand);
+            SubscribeLocalEvent<WieldableComponent, ItemWieldedEvent>(OnItemWielded);
+            SubscribeLocalEvent<WieldableComponent, ItemUnwieldedEvent>(OnItemUnwielded);
+            SubscribeLocalEvent<WieldableComponent, RemovedFromHandEvent>(OnItemLeaveHand);
+            SubscribeLocalEvent<WieldableComponent, VirtualItemDeletedEvent>(OnVirtualItemDeleted);
+
+            SubscribeLocalEvent<IncreaseDamageOnWieldComponent, MeleeHitEvent>(OnMeleeHit);
+        }
+
+        private void OnUseInHand(EntityUid uid, WieldableComponent component, UseInHandEvent args)
+        {
+            if (args.Handled)
+                return;
+            if(!component.Wielded)
+                AttemptWield(uid, component, args.User);
+            else
+                AttemptUnwield(uid, component, args.User);
+        }
```

Comment:
I think at the moment this clashes with the functionality to equipt axes & other items to the backpack slot. When you left click or use z this starts the do-after, but then it also equips the axe to the backpack slot causing the do-after to fail because the item is no longer in your hands.

Currently it seems to work by just setting `args.handled = true` at the end of this function. But AFAIK the UseInHandEvent doesn't really have a priority system, so I guess it's just a coincidence that this is run before the code that equips the axe.

Reply:
you -can- order events using a topographical sort, i just cant use that right now because equipment UseInHand goes through a comp interface and not an event, so I'd have to change that

Reply:
decided to just set QuickEquip false on clothing for now